### ISSUE TITLE
why is srun needed?

### ISSUE THREAD

**Issue by hvgazula** [sabeenlohawala/tissue_labeling#41]
just curious..why is `srun` needed here?

https://github.com/sabeenlohawala/tissue_labeling/blob/f88c8a08ccb72e2c4cde1867337b2d46f7e64556/submit_multi_gpu.sh#L20

**Comment by sabeenlohawala** [comment]
We looked into this in December I think, but srun is used to run a parallel job: https://slurm.schedmd.com/srun.html.

**Comment by hvgazula** [comment]
but there's only one job..my submit.sh doesn't have srun..does it?

**Comment by hvgazula** [comment]
also any reason why the arguments are placed inside the submission script but not outside?https://github.com/sabeenlohawala/tissue_labeling/blob/dev/submit_requeue.sh

**Comment by hvgazula** [comment]
if requeue is for resumption..those parameters can be read from config file..correct? or is It that you except requeue for train as well?

**Comment by sabeenlohawala** [comment]
> but there's only one job..my submit.sh doesn't have srun..does it?

Yes, there's only one job but because the job uses multiple gpus the srun command is used to create the multiple processes (from what I understand). Your submit.sh doesn't have it, but Matthias's did. In December, we said that I should modify the submit.sh to have the srun command.

> also any reason why the arguments are placed inside the submission script but not outside?https://github.com/sabeenlohawala/tissue_labeling/blob/dev/submit_requeue.sh

I did this to make sure that if the job was requeued, the correct arguments would still be passed so the correct logdir and its checkpoints could be found if the job is resumed. If the arguments are passed through the command line / makefile to the .sh script are they remembered when they're requeued?

> if requeue is for resumption..those parameters can be read from config file..correct? or is It that you except requeue for train as well?

Yes, if the job is requeued it just finds the latest checkpoint and reads in the arguments from the config, but the naming convention of the logdir depends on the values for these arguments.

**Comment by hvgazula** [comment]
https://github.com/MGH-LEMoN/Photo-SynthSeg/blob/fixed-spacing/Makefile#L370

**Comment by hvgazula** [comment]
Now I understand why I have `train` and `resume-train`. If only I knew about the `--requeue` I would have combined the two flags by modifying the [`update_config`](https://github.com/sabeenlohawala/tissue_labeling/blob/0a22ee0ca670414ddff090d12828822cd5d097ae/scripts/commands/main.py#L54-L87) function to work on the existence of a `logdir` instead of the flag (`train`|`retrain`). Anyway, good to know. Thanks.
